File: Cours/AP1/CH4/TD22_bis/etat-civil.c
#include <stdint.h>
#include <stdbool.h>
#include <stdio.h>

typedef char *chaine_t;


typedef enum {
    JANVIER = 1, FEVRIER, MARS, AVRIL, MAI, JUIN, JUILLET, AOUT, SEPTEMBRE, OCTOBRE, NOVEMBRE, DECEMBRE
} mois_t;

typedef struct {
    uint8_t jour;
    mois_t mois;
    int annee;
} date_t;

typedef struct {
    chaine_t nomPrenom;
    date_t naissance;
    double taille;
    bool motard;
} infos_t;


typedef struct {
    int entiere;
    int fract;
} resultat_modf_t;

resultat_modf_t ma_modf(double val) {
    resultat_modf_t res;
    res.entiere = (int)val;
    res.fract = (int)((val - res.entiere) * 100);
    return res;
}

void afficher(infos_t moi){
    printf("Je m'apelle %s \n", moi.nomPrenom);
    printf("Je suis né le %d/%d/%d \n", moi.naissance.jour, moi.naissance.mois, moi.naissance.annee);
    printf("Je mesure %g \n", moi.taille);

    if (moi.motard){
        printf("GAAAAAAZ \n ");
    }else{
        printf("C'est quoi une moto ? \n");
    }
}

int main(void) {
    infos_t eddy = {"Édouard Bracame", {15, JUIN, 1950}, 1.78, true};
    infos_t obelix = {.taille = 1.83, .naissance = {4, SEPTEMBRE, -70}, .nomPrenom = "Obelix" };
    obelix.motard = false;
    afficher(eddy);
    puts("");
    afficher(obelix);
}

File: Cours/AP1/CH1/TD2/etat_civil.go
package main

import (
	"math"
	"strconv"
)

type lesMois int64

const (
	janvier lesMois = iota + 1
	fevrier
	mars
	avril
	mais
	juin
	juillet
	aout
	septembre
	octobre
	novembre
	decembre
)

type date struct {
	jour  int8
	mois  lesMois
	annee uint
}

type info struct {
	prenom    string
	nom       string
	naissance date
	taille    float64
	motarde   bool
}

func main() {

	gens1 := info{
		prenom: "Bastien",
		nom:    "Pichet",
		naissance: date{
			jour:  22,
			mois:  juillet,
			annee: 2005,
		},

		taille:  1.70,
		motarde: false,
	}

	gens2 := info{
		prenom: "Chloe",
		nom:    "Ramon",
		naissance: date{
			jour:  11,
			mois:  mars,
			annee: 2005,
		},

		taille:  1.70,
		motarde: false,
	}

	affichage(gens1)
	affichage(gens2)
}

func affichage(gens info) {

	println("Je m'apelle", gens.prenom, gens.nom)
	println(strconv.Itoa(int(gens.naissance.jour)) + "/" + strconv.Itoa(int(gens.naissance.mois)) + "/" + strconv.Itoa(int(gens.naissance.annee)))
	var entier float64
	var fraction float64
	entier, fraction = math.Modf(gens.taille)
	fraction = fraction * 100
	println("Je mesure", strconv.Itoa(int(entier)), "metre", strconv.Itoa(int(fraction)))
}
